forui/lib/src/widgets/accordion.dart

```diff
@@ -0,0 +1,525 @@
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+import 'package:forui/src/foundation/tappable.dart';
+import 'package:forui/src/foundation/util.dart';
+
+/// A controller that stores the expanded state of an [_Item].
+interface class FAccordionController extends ChangeNotifier {
+  /// The duration of the expansion and collapse animations.
+  final Duration duration;
+  final List<(AnimationController, Animation)> _controllers;
+  final Set<int> _values;
+  final int _min;
+  final int? _max;
+
+  /// Creates a [FAccordionController].
+  ///
+  /// The [min] and [max] values are the minimum and maximum number of selections allowed. Defaults to no minimum or maximum.
+  ///
+  /// # Contract:
+  /// * Throws [AssertionError] if [min] < 0.
+  /// * Throws [AssertionError] if [max] < 0.
+  /// * Throws [AssertionError] if [min] > [max].
+  FAccordionController({
+    int min = 0,
+    int? max,
+    Set<int>? values,
+    this.duration = const Duration(milliseconds: 500),
+  })  : _min = min,
+        _max = max,
+        _controllers = [],
+        _values = values ?? {},
+        assert(min >= 0, 'The min value must be greater than or equal to 0.'),
+        assert(max == null || max >= 0, 'The max value must be greater than or equal to 0.'),
+        assert(max == null || min <= max, 'The max value must be greater than or equal to the min value.');
+
+  /// Adds an item to the accordion.
+  void addItem(int index, AnimationController controller, Animation expand) {
+    _controllers.add((controller, expand));
+    if (controller.value == 1.0) {
+      _values.add(index);
+    }
```

Comment:
This logic seems incorrect. What if you:
* Add an item at an index that already exists.
* Add an item at an index that isn't at the end of the list?

Reply:
Why do we only add an index to `_values` if the animation == 1, while we unconditionally add both to the list?